tests/unit/test_client_options.py

```diff
@@ -144,10 +144,20 @@ def test_from_dict_bad_argument():
 
 
 def test_repr():
+    expected_keys = set(
+        [
+            "api_endpoint",
+            "client_cert_source",
+            "client_encrypted_cert_source",
+            "quota_project_id",
+            "credentials_file",
+            "scopes",
+            "api_key",
+            "api_audience",
+        ]
+    )
     options = client_options.ClientOptions(api_endpoint="foo.googleapis.com")
+    options_keys = vars(options).keys()
 
-    assert (
-        repr(options)
-        == "ClientOptions: {'api_endpoint': 'foo.googleapis.com', 'client_cert_source': None, 'client_encrypted_cert_source': None, 'api_key': None}"
-        or "ClientOptions: {'client_encrypted_cert_source': None, 'client_cert_source': None, 'api_endpoint': 'foo.googleapis.com', 'api_key': None}"
-    )
+    assert options_keys == expected_keys
+    assert options.api_endpoint == "foo.googleapis.com"
```

Comment:
Should we explicitly test that `client_encrypted_cert_source` and `client_cert_source` are `None`, like the previous version was doing? I guess that would verify the constructor does not auto-populate some keys.

Reply:
Done.